docs/DeveloperGuide.md

```diff
@@ -1,12 +1,186 @@
 # Developer Guide
 
 ## Acknowledgements
+Our project uses external libraries and services from:
+1. Land Transport Authority DataMall API Service ([link](https://datamall.lta.gov.sg/content/datamall/en.html)).
+2. Jackson JSON Parser ([link](https://fasterxml.github.io/jackson-core/javadoc/2.8/com/fasterxml/jackson/core/JsonParser.html))
+
+## Design 
+
+### Architecture Level
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+### Component Level
```

Comment:
Parameter-list for methods should be of the form [name of variable] : [type of variable] e.g. setNumber(n: Integer)